packages/core/src/parsers/jsx.ts

```diff
@@ -908,6 +917,16 @@ function undoPropsDestructure(
       return pre;
     }, {} as Record<string, string>);
 
+    if (param.typeAnnotation) {
+      node.params = [
+        {
+          ...babel.types.identifier('props'),
+          typeAnnotation: param.typeAnnotation,
+        } as babel.types.Identifier,
```

Comment:
This type-cast is unnecessary and can be removed

```suggestion
        },
```